Empire/EmpireManager.cpp

```diff
@@ -226,6 +238,9 @@ void EmpireManager::HandleDiplomaticMessage(const DiplomaticMessage& message) {
     int sender_empire_id = message.SenderEmpireID();
     int recipient_empire_id = message.RecipientEmpireID();
 
+    if (GetGameRules().Get<std::string>("RULE_DIPLOMACY") == "RULE_DIPLOMACY_FORBIDDEN_FOR_ALL")
```

Comment:
String literals that are used as stringtable entry keys, if not being passed as a parameter to a function, should generally be wrapped in `UserStringNoOp` (or whatever it is, I can't check the spelling right now)

Reply:
Wrapped.